zerver/lib/actions.py

```diff
@@ -3799,11 +3799,26 @@ def do_update_message(user_profile: UserProfile, message: Message, topic_name: O
     }  # type: Dict[str, Any]
     changed_messages = [message]
 
+    send_event_users = list()  # call update event for these users
+    ums = UserMessage.objects.filter(message=message.id)
+    ums_user_ids = set(user.user_profile_id for user in ums)
     if message.is_stream_message():
         stream_id = message.recipient.type_id
         event['stream_name'] = Stream.objects.get(id=stream_id).name
-
-    ums = UserMessage.objects.filter(message=message.id)
+        # Get subscriber ids of the stream who are active now.
+        active_subscriber_ids = set(get_active_subscriptions_for_stream_id(
+            stream_id).values_list("user_profile_id", flat=True))
+        # Also send update event to subscribers who are not
+        # not in ums, i.e. they subscribed to the stream after
+        # the message was sent, or are not subscribed at all.
+        # TODO: Send update event to active non-subscribers who
+        # can see the message.
```

Comment:
I would replace this TODO with:

"Note: We don't send the update event to users who are not subscribed to this stream and don't have a UserMessage row.  This means if a non-subscriber is viewing the narrow, they won't get real-time updates.  This is a reasonable balance between sending message-edit notifications for every public stream to every user (too expensive, and also not what we do for newly sent messages anyway) and having magical live-updates where possible."